packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -334,6 +342,42 @@ abstract class DoctorValidator {
   Future<ValidationResult> validate();
 }
 
+/// A compatibility layer until all custom doctor implementations use the
+/// extension shim.
+class ExtensionCompatDoctorValidation implements DoctorValidator {
```

Comment:
I think we'll need to change this:

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/doctor.dart#L252

From using runtimeType for the validator name to something else.